RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -35,9 +34,9 @@ public Mono<Void> reindex(
                 bulkJson -> client.sendBulkRequest(indexName, bulkJson, context.createBulkRequest()) // Send the request
                     .doOnSuccess(unused -> logger.debug("Batch succeeded"))
                     .doOnError(error -> logger.error("Batch failed", error))
-                    .onErrorResume(e -> Mono.empty()) // Prevent the error from stopping the entire stream
-            )
-            .retryWhen(Retry.backoff(3, Duration.ofSeconds(1)).maxBackoff(Duration.ofSeconds(5)))
+                    // Prevent the error from stopping the entire stream, retries occurring within sendBulkRequest
```

Comment:
@AndreKurait Lets add a test to confirm the expected behavior, recently I created a PR that verified a similar scenario [1].

- [1] https://github.com/opensearch-project/opensearch-migrations/pull/882

Reply:
I'd recommend testing this because we previously had a bug around this logic that required this change.

Reply:
> Note: @gregschohn will take care of unit tests for this stuff today